core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -177,12 +180,31 @@ fn consume_scan_should_process_packet(
             return ProcessingDecision::Never;
         }
 
+        if !payload.account_locks.check_locks(message) {
```

Comment:
did a separate `check_locks`, then `take_locks` so that we will not check fee-payers that we know can't even be in this batch.